src/components/KeyboardShortcutsModal.js

```diff
@@ -39,30 +39,53 @@ class KeyboardShortcutsModal extends React.Component {
-        this.unsubscribeCloseEnterModal = KeyboardShortcut.subscribe(closeShortcutEnterModalConfig.shortcutKey, () => {
-            ModalActions.close();
-            KeyboardShortcutsActions.hideKeyboardShortcutModal();
-        }, closeShortcutEnterModalConfig.descriptionKey, closeShortcutEnterModalConfig.modifiers, true, () => !this.props.isShortcutsModalOpen);
+            const closeShortcutEnterModalConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
+            this.unsubscribeCloseEnterModal = KeyboardShortcut.subscribe(closeShortcutEnterModalConfig.shortcutKey, () => {
+                ModalActions.close();
+                KeyboardShortcutsActions.hideKeyboardShortcutModal();
+            }, closeShortcutEnterModalConfig.descriptionKey, closeShortcutEnterModalConfig.modifiers, true);
 
-        // Intercept arrow up and down keys to prevent scrolling ArrowKeyFocusManager while this modal is open
-        const arrowUpConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_UP;
-        this.unsubscribeArrowUpKey = KeyboardShortcut.subscribe(arrowUpConfig.shortcutKey, () => {
-        }, arrowUpConfig.descriptionKey, arrowUpConfig.modifiers, true, () => !this.props.isShortcutsModalOpen);
+            // Intercept arrow up and down keys to prevent scrolling ArrowKeyFocusManager while this modal is open
+            const arrowUpConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_UP;
+            this.unsubscribeArrowUpKey = KeyboardShortcut.subscribe(arrowUpConfig.shortcutKey, () => {
+            }, arrowUpConfig.descriptionKey, arrowUpConfig.modifiers, true);
 
-        const arrowDownConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_DOWN;
-        this.unsubscribeArrowDownKey = KeyboardShortcut.subscribe(arrowDownConfig.shortcutKey, () => {
-        }, arrowDownConfig.descriptionKey, arrowDownConfig.modifiers, true, () => !this.props.isShortcutsModalOpen);
+            const arrowDownConfig = CONST.KEYBOARD_SHORTCUTS.ARROW_DOWN;
+            this.unsubscribeArrowDownKey = KeyboardShortcut.subscribe(arrowDownConfig.shortcutKey, () => {
+            }, arrowDownConfig.descriptionKey, arrowDownConfig.modifiers, true);
+        } else if (prevProps.isShortcutsModalOpen && !this.props.isShortcutsModalOpen) {
+            // Modal is closing, remove keyboard shortcuts
+            if (this.unsubscribeCloseEscapeModal) {
+                this.unsubscribeCloseEscapeModal();
+                this.unsubscribeCloseEscapeModal = undefined;
+            }
+            if (this.unsubscribeCloseEnterModal) {
+                this.unsubscribeCloseEnterModal();
+                this.unsubscribeCloseEnterModal = undefined;
+            }
+            if (this.unsubscribeArrowUpKey) {
+                this.unsubscribeArrowUpKey();
+                this.unsubscribeArrowUpKey = undefined;
+            }
+            if (this.unsubscribeArrowDownKey) {
+                this.unsubscribeArrowDownKey();
+                this.unsubscribeArrowDownKey = undefined;
+            }
```

Comment:
Agreed, I thought of creating a class component that would be rendered inside the modal only when `prop.isShortcutsModalOpen = true`.  That way we could have used mount/unmount which is a bit cleaner I think.

I didn't continue because we are supposed to migrate everything to hooks, so it would have to be redone anyway.